### DISCUSSION TITLE
Unshallow Bug/Endless Loop

### DISCUSSION THREAD

**Discussion by DctrNoob** [Homebrew/discussions#266] [Everyday usage] [answered]
After calling the unshallow command brew still complains about the need to unshallow:

% git -C "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core" fetch --unshallow
remote: Enumerating objects: 413087, done.
remote: Counting objects: 100% (413058/413058), done.
remote: Compressing objects: 100% (161190/161190), done.
remote: Total 405032 (delta 245914), reused 400145 (delta 241032), pack-reused 0
Receiving objects: 100% (405032/405032), 138.84 MiB | 1010.00 KiB/s, done.
Resolving deltas: 100% (245914/245914), completed with 3752 local objects.

% brew update
Error: homebrew-cask is a shallow clone. To `brew update` first run:
  git -C "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask" fetch --unshallow
This restriction has been made on GitHub's request because updating shallow
clones is an extremely expensive operation due to the tree layout and traffic of
Homebrew/homebrew-cask. We don't do this for you automatically to avoid
repeatedly performing an expensive unshallow operation in CI systems (which
should instead be fixed to not use shallow clones). Sorry for the inconvenience!

% git -C "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core" fetch --unshallow
fatal: --unshallow on a complete repository does not make sense

**Comment by carlocab** [comment, marked as answer]
You also need to do 
```
git -C "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask" fetch --unshallow
```
Note that it says homebrew-**cask**, note homebrew-core.

**Reply by DctrNoob** [reply]
Ah, thanks! I missed that. That did it.

**Reply by carlocab** [reply]
I missed it too on a first reading. Glad it worked for you.

**Comment by mbierman** [comment]
I had the same question and this solved it... but I am curious, is "--unshallow" a new requirement? Why am I just seeing this now for the first time?

**Reply by SMillerDev** [reply]
See https://github.com/Homebrew/discussions/discussions/226

**Comment by rutledgeismyname** [comment]
So I see this message

"This restriction has been made on GitHub's request because updating shallow
clones is an extremely expensive operation due to the tree layout and traffic of
Homebrew/homebrew-core. We don't do this for you automatically to avoid
repeatedly performing an expensive unshallow operation in CI systems (which
should instead be fixed to not use shallow clones). Sorry for the inconvenience!"

Do I have to actually take any action on it? It kinda worries me. 

@carlocab

**Reply by gromgit** [reply]
That just explains why shallow clones are now banned in Homebrew. Unless you're using shallow clones in your CI setup, `--unshallow` should be all you need to do.
